pkg/queryengines/mongoqueryengine/connection.go

```diff
@@ -21,7 +22,14 @@ func createMongoConnectionURI(scheme string, host string, port uint16, user, pas
 		usernamePassword = user + ":" + password + "@"
 	}
 	if scheme == "mongodb" {
-		return "mongodb://" + usernamePassword + host + ":" + strconv.Itoa(int(port))
+		// Adding support to connect to Azure CosmosDB using MongoDB API.
+		// According to official docs, the connection string should pass
+		// ssl=true param to connect.
+		if strings.Contains(host, ".mongo.cosmos.azure.com") {
+			return "mongodb://" + usernamePassword + host + ":" + strconv.Itoa(int(port)) + "/?ssl=true"
```

Comment:
could this be given as checkbox option rather than detecting host contains string?

It will probably enable other hosts like AWS or GCP too?

Reply:
Yes, it can be done. I will make changes and update the PR.

Reply:
New code changes merged.